android/app/src/main/kotlin/com/jabook/app/jabook/audio/PlaybackEventProcessor.kt

```
@@ -0,0 +1,283 @@
+        }
+
+        // Optimize memory for large playlists
+        if (totalTracks > 50) {
+            optimizeMemoryUsage?.invoke(currentIndex)
+        }
+
+        // Restart position check on last track while playing
+        if (player.isPlaying &&
+            !getIsBookCompleted() &&
+            player.playbackState == Player.STATE_READY &&
+            currentIndex >= totalTracks - 1
```

detekt (1.23.8)

[warning] This condition is too complex (4). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)